src/tools/perf/perftest_daemon.c

```diff
@@ -0,0 +1,511 @@
+/**
+ * Copyright (C) NVIDIA 2022.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <ucp/api/ucp.h>
+#include <ucs/debug/log.h>
+#include <ucs/type/serialize.h>
+#include <tools/perf/api/libperf.h>
+#include <string.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <signal.h>
+
+
+typedef struct ucp_perf_daemon_context_t {
+    ucp_context_h                      context;
+    ucp_worker_h                       worker;
+    ucp_listener_h                     listener;
+    uint16_t                           port;
+    ucp_ep_h                           client_ep;
+    ucp_ep_h                           peer_ep;
+    struct sockaddr_storage            peer_address;
+    ucp_mem_h                          send_memh;
+    ucp_mem_h                          recv_memh;
+    void                               *rx_address;
+} ucp_perf_daemon_context_t;
+
+static volatile int terminated = 0;
```

Comment:
This is rather a question than a comment:
Do we have a policy for naming global variables? Smth like g_terminated?

Reply:
As I know, we have such rules only for global variables inside the UCX modules (they should be prefixed by the module name e.g. `ucp_atomic_modes`).